src/framework/mlt_chain.c

```diff
@@ -393,6 +394,56 @@ void mlt_chain_close( mlt_chain self )
 	}
 }
 
+/** Attach normalizer links.
+ *
+ * \public \memberof mlt_chain_s
+ * \param self a chain
+ */
+
+extern void mlt_chain_attach_normalizers( mlt_chain self )
+{
+	if ( !self ) return;
+
+	if ( mlt_chain_link_count( self ) && mlt_properties_get_int( MLT_LINK_PROPERTIES( mlt_chain_link( self, 0 ) ), "_loader" ) )
+	{
+		// Chain already has normalizers
+		return;
+	}
+
+	static mlt_properties normalisers = NULL;
+
+	mlt_tokeniser tokenizer = mlt_tokeniser_init( );
+
+	// We only need to load the normalising properties once
+	if ( normalisers == NULL )
+	{
+		char temp[ 1024 ];
+		sprintf( temp, "%s/chain_normalizers.ini", mlt_environment( "MLT_DATA" ) );
+		normalisers = mlt_properties_load( temp );
+		mlt_factory_register_for_clean_up( normalisers, ( mlt_destructor )mlt_properties_close );
+	}
+
+	// Apply normalisers
+	for ( int i = 0; i < mlt_properties_count( normalisers ); i ++ )
+	{
+		char *value = mlt_properties_get_value( normalisers, i );
+		mlt_tokeniser_parse_new( tokenizer, value, "," );
+		for ( int j = 0; j < mlt_tokeniser_count( tokenizer ); j ++ )
+		{
+			mlt_link link = mlt_factory_link( mlt_tokeniser_get_string( tokenizer, j ), NULL );
+			if ( link )
+			{
+				mlt_properties_set_int( MLT_LINK_PROPERTIES(link), "_loader", 1 );
+				mlt_chain_attach( self, link );
+				mlt_chain_move_link( self, mlt_chain_link_count( self ) - 1, 0 );
```

Comment:
Please be aware that this move link call means that links will be added in the reverse order from the ini file. That is not a big deal and never may be. Feel free to ignore it for now, but FYI.

Reply:
Good comment. Noted.